docs/getting-started-tutorial/3-adding-snippet-bindings.adoc

```diff
@@ -0,0 +1,73 @@
+:idprefix:
+:idseparator: -
+:toc: right
+:toclevels: 2
+
+# Adding Snippet Bindings
+
+In most frameworks, a page's data is looked up by a controller, and backend
+code clutters the HTML to produce the correct rendering of the data. This
+process is usually done through what amounts to little more than string
+munging. Lift throws this paradigm away entirely in favor of a much better
+approach based on entities called snippets.
+
+Snippets let you refer to a block of code that is responsible for rendering a
+particular part of your page. You add these references by augmenting your HTML
+with a few completely valid `data-` attributes that get stripped before the
+HTML is then sent to the browser. These snippets then take your HTML, fully
+parsed into a valid DOM tree, and transform it, providing true decoupling
+between your business logic and your template, and an extra level of
+security footnote:[We already mentioned that Lift is secure by default, and
+another way that manifests is that the template HTML is turned into a
+first-class XML tree early in the processing cycle, and snippets just transform
+that tree. That means script injection and a variety of other attacks are
+significantly more difficult against a Lift codebase.].
+
+
+Let's look at our chat app specifically. We're going to bind two things: the
+list of chat messages, and the text input that lets us actually chat. To the
+`ol` that contains the chat messages, we add:
+
+```
+<ol class="messages" data-lift="Chat.messages">
+```
+
+And to the input form:
+
+```
+<form class="send-message" data-lift="Chat.sendMessage">
+```
+
+These two indicate two methods in a class called `Chat`, which Lift searches
+for in the `code.snippet` package footnote:[This can be changed using
+link:++https://liftweb.net/api/30/api/index.html#net.liftweb.http.LiftRules@addToPackages(what:String):Unit++[`LiftRules.addPackage`.].
+We'll write a very basic version that just passes through the contents of the
+list and form unchanged, and then in the next section we'll start adding some
+behavior. In `src/main/scala/code/snippet/Chat.scala`, add:
+
+```
+package code
+package snippet
+
+import scala.xml._
+
+object Chat {
+  def messages(contents: NodeSeq) = contents
+  def sendMessage(contents: NodeSeq) = contents
+}
```

Comment:
It would be really cool at some point in the future to take a shot at integrating dexy so all of these have to compile or something. But that's not for now. =)

Reply:
100% part of my plan, yeah.